cpp/src/arrow/ipc/adapter.cc

```diff
@@ -226,15 +252,17 @@ class RowBatchReader::Impl {
     // we can skip that buffer without reading from shared memory
     FieldMetadata field_meta = metadata_->field(field_index_++);
 
+    // extract null_bitmap which is common to all arrays
+    std::shared_ptr<Buffer> null_bitmap;
+    if (field_meta.null_count == 0) {
+      null_bitmap = nullptr;
```

Comment:
I don't think so.  The way the spec is currently worded it says it is optional on whether null_count == 0 implies a null pointer, and that the presence of it should be communicated in the metadata.  I need to check Message.fbs, but I'm not sure it currently allows for communicating whether the bitmap is included.  Here we make the decision to never include the null bitmap because the data-structure is redundant.  I will add a comment explaining this.